public/app/features/dashboard/components/PanelEditor/PanelEditorTabs.tsx

```diff
@@ -16,55 +16,77 @@ interface PanelEditorTabsProps {
   onChangeTab: (tab: PanelEditorTab) => void;
 }
 
-export const PanelEditorTabs: React.FC<PanelEditorTabsProps> = ({ panel, dashboard, tabs, onChangeTab }) => {
-  const styles = getPanelEditorTabsStyles();
-  const activeTab = tabs.find(item => item.active)!;
+export class PanelEditorTabs extends PureComponent<PanelEditorTabsProps> {
+  componentDidMount() {
+    const { panel } = this.props;
+    panel.on(CoreEvents.queryAdded, this.triggerForceUpdate);
+    panel.on(CoreEvents.queryRemoved, this.triggerForceUpdate);
+    panel.on(CoreEvents.transformationAdded, this.triggerForceUpdate);
+    panel.on(CoreEvents.transformationRemoved, this.triggerForceUpdate);
```

Comment:
Wonder if having `queriesChanged`/`transformationsChanged` events wouldn't be sufficient? I suggest this as we are using these events only in this single case, and limiting the number of events would simplify the logic responsible for actually sending those. You would avoid checking whether or not something was added/removed. I think the less code in this case the better:)

Reply:
Hm, yes that's true. Updating

Reply:
thank you!

Reply:
Great suggestion @dprokop

Reply:
It seems like the CI agreed! :smile:

Reply:
hahah, yeah!